docs/core_docs/docs/tutorials/chatbot.ipynb

```diff
@@ -1,1149 +1,1448 @@
-      ]
-    },
+ "cells": [
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "vscode": {
+     "languageId": "raw"
+    }
+   },
+   "source": [
+    "---\n",
+    "sidebar_position: 1\n",
+    "keywords: [conversationchain]\n",
+    "---"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Build a Chatbot\n",
+    "\n",
+    "\n",
+    ":::info Prerequisites\n",
+    "\n",
+    "This guide assumes familiarity with the following concepts:\n",
+    "\n",
+    "- [Chat Models](/docs/concepts/chat_models)\n",
+    "- [Prompt Templates](/docs/concepts/prompt_templates)\n",
+    "- [Chat History](/docs/concepts/chat_history)\n",
+    "\n",
+    "This guide requires `langgraph >= 0.2.28`.\n",
+    "\n",
+    ":::\n",
+    "\n",
+    "\n",
+    "```{=mdx}\n",
+    "\n",
+    ":::note\n",
+    "\n",
+    "This tutorial previously built a chatbot using [RunnableWithMessageHistory](https://api.js.langchain.com/classes/_langchain_core.runnables.RunnableWithMessageHistory.html). You can access this version of the tutorial in the [v0.2 docs](https://js.langchain.com/v0.2/docs/tutorials/chatbot/).\n",
+    "\n",
+    "The LangGraph implementation offers a number of advantages over `RunnableWithMessageHistory`, including the ability to persist arbitrary components of an application's state (instead of only messages).\n",
+    "\n",
+    ":::\n",
+    "\n",
+    "```\n",
+    "\n",
+    "## Overview\n",
+    "\n",
+    "We'll go over an example of how to design and implement an LLM-powered chatbot. \n",
+    "This chatbot will be able to have a conversation and remember previous interactions.\n",
+    "\n",
+    "\n",
+    "Note that this chatbot that we build will only use the language model to have a conversation.\n",
+    "There are several other related concepts that you may be looking for:\n",
+    "\n",
+    "- [Conversational RAG](/docs/tutorials/qa_chat_history): Enable a chatbot experience over an external source of data\n",
+    "- [Agents](https://langchain-ai.github.io/langgraphjs/tutorials/multi_agent/agent_supervisor/): Build a chatbot that can take actions\n",
+    "\n",
+    "This tutorial will cover the basics which will be helpful for those two more advanced topics, but feel free to skip directly to there should you choose.\n",
+    "\n",
+    "## Setup\n",
+    "\n",
+    "### Jupyter Notebook\n",
+    "\n",
+    "This guide (and most of the other guides in the documentation) uses [Jupyter notebooks](https://jupyter.org/) and assumes the reader is as well. Jupyter notebooks are perfect for learning how to work with LLM systems because oftentimes things can go wrong (unexpected output, API down, etc) and going through guides in an interactive environment is a great way to better understand them.\n",
+    "\n",
+    "This and other tutorials are perhaps most conveniently run in a Jupyter notebook. See [here](https://jupyter.org/install) for instructions on how to install.\n",
```

Comment:
Can be in a followup but we should have a guide on how to set up TSLab since it's not a typical experience for JS devs

Reply:
Agree